concepts/observability-configurable-monitoring/01-groundwork.md

```diff
@@ -0,0 +1,59 @@
+  - Support typical settings for the supported outputs in a convenient way
+  - Non-typical settings should be possible as well, leaving the agent in an unsupported mode
+  - Secrets must be kept secret.
+  - Secret bindings for the SAP BTP Operator must be supported incl. secret rotations
+  - Scenarios/Pipelines need to be isolated and have it's own buffer management. If a backend is in a bad shape and cannot process any data anymore, data should still continue to be pushed to other backends
+  - Typical auth mechanisms for the integration must be support, especially also client certificate based solutions
+- Inputs
+  - Inputs are pre-defined and should be based on the pull and push-based approach. In best case the user do not need to differentiate between the two types as they bring no semantical difference
+  - The input should be filterable on namespace/workload level, metrics of uninteresting components should not create a resource footprint in the pipeline in best case
+- Filter
+  - ing of data (like dropping metrics of kyma-system components) must be possible
+
+## Pre-integration
+- Kyma system components are pre-integrated, so the pre-defined input in the agent serves them by default
+- Typical k8s metrics are pre-integrated, including basic node and kube-state-metrics. Users might need this data for own troubleshooting.
+- Istio metrics are pre-integrated and can be easily filtered and de-selected
+
+## Ease of integration
+- It should be trivial to integrate metrics of a workload. Either by using a push approach to a well known internal URL in a vendor-neutral protocol, or by a pull approach by annotating the workload.
+- Envoy metrics should get collected instantly (without user action)
+
+## Local Backend
+- Installing the monitoring component ships managed prometheus instances
+- The setup is not meant to be HA and scalable
+- The setup should handle over-load properly so that it is always in a functional state
+- Multiple instances might be used for the different scenarios like system, istio and custom. That will assure reliability of one scenario if the other is in trouble.
+
+# General Idea
+
+The idea of the proposal is to introduce a preconfigured agent layer that's responsible for collecting metric data only. Those agent can be configured dynamically at runtime with different configuration scenarios, so that the agents start shipping the data to the configured backends. The dynamic configuration and management of the agent is handled by a new operator, which is configured using Kubernetes APIs. The agent and the new operator are bundled in a new core package called `telemetry`. The existing Kyma backends and UIs will be just one possible solution that can be installed optionally and will stay in the existing modules like `monitoring`.
```

Comment:
```suggestion
The idea of the proposal is to introduce a preconfigured agent layer that's responsible for collecting metric data only. Those agents can be configured dynamically at runtime with different configuration scenarios, so that the agents start shipping the data to the configured backends. The dynamic configuration and management of the agent is handled by a new operator, which is configured using Kubernetes APIs. The agent and the new operator are bundled in a new core package called `telemetry`. The existing Kyma backends and UIs will be just one possible solution that can be installed optionally and will stay in the existing modules like `monitoring`.
```
Excellent writing! Nearly nothing I could nitpick here!